ouroboros-network/protocol-tests/Ouroboros/Network/Protocol/LocalStateQuery/Test.hs

```diff
@@ -321,26 +368,32 @@ prop_codec
   :: AnyMessageAndAgency (LocalStateQuery Block (Point Block) Query)
   -> Bool
 prop_codec msg =
-  runST (prop_codecM codec msg)
+  runST (prop_codecM (codec True) msg)
 
 -- | Check for data chunk boundary problems in the codec using 2 chunks.
 --
 prop_codec_splits2
   :: AnyMessageAndAgency (LocalStateQuery Block (Point Block) Query)
   -> Bool
 prop_codec_splits2 msg =
-  runST (prop_codec_splitsM splits2 codec msg)
+  runST (prop_codec_splitsM splits2 (codec True) msg)
 
 -- | Check for data chunk boundary problems in the codec using 3 chunks.
 --
 prop_codec_splits3
   :: AnyMessageAndAgency (LocalStateQuery Block (Point Block) Query)
   -> Bool
 prop_codec_splits3 msg =
-  runST (prop_codec_splitsM splits3 codec msg)
+  runST (prop_codec_splitsM splits3 (codec True) msg)
 
 prop_codec_cbor
   :: AnyMessageAndAgency (LocalStateQuery Block (Point Block) Query)
   -> Bool
 prop_codec_cbor msg =
-  runST (prop_codec_cborM codec msg)
+  runST (prop_codec_cborM (codec True) msg)
+
+prop_codec_V7_compatible
+  :: AnyMessageAndAgencyV7
+  -> Bool
+prop_codec_V7_compatible (AnyMessageAndAgencyV7 msg) =
+    runST (prop_codecs_compatM (codec False) (codec True) msg)
```

Comment:
Yep nice.

Perhaps more code duplication in the `AnyMessageAndAgencyV7` QC instance than necessary, but ok.

Reply:
I should  write them them the other way around, its easier to extend an instance than shrink.

Reply:
Fixed.